### PR TITLE
EPMLSTRCMW-290 Replace mocks with test implementations in service tests

### PR DESCRIPTION
**Pull request by Nikolai-Isaev**
test: Replace mocks with impls in service

### CODE REVIEW

File: auth/src/main/scala/cromwell/pipeline/auth/AuthUtils.scala [hunk 1]
```diff
@@ -5,57 +5,73 @@ import cromwell.pipeline.utils.AuthConfig
...
-  }
+          AuthResponse(newAccessToken, newRefreshToken, currentTimestamp + updatedAccessTokenLifetime)
+        }
 
+        refreshTokenContent.optRestOfUserSession match {
+          case optRestOfUserSession if optRestOfUserSession.exists(_ > 0) || optRestOfUserSession.isEmpty =>
+            Some(buildAuthResponse)
+          case _ =>
+            None
+        }
```
Comment (DruzhininPavel) [inline auth/src/main/scala/cromwell/pipeline/auth/AuthUtils.scala:61]: I think this could be changed for more understanding. But it's up to you to decide.
```suggestion
        refreshTokenContent.optRestOfUserSession match {
          case optRestOfUserSession if optRestOfUserSession.exists(_ <= 0) => None
          case _                                                                                                   => Some(buildAuthResponse)
        }
```

**Review by DruzhininPavel — APPROVED** [review]
Great job, I just have one comment.

**Review by gominasian — APPROVED** [review]

File: auth/src/main/scala/cromwell/pipeline/auth/AuthUtils.scala (hunk 1 above)
Comment (Nikolai-Isaev) [reply auth/src/main/scala/cromwell/pipeline/auth/AuthUtils.scala:61]: I prefer not to change code which is not directly involved in the task

File: services/src/test/scala/cromwell/pipeline/service/impls/WomToolTestImpl.scala [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        getMessages
+
+      override def stringInputs(content: String): Either[NonEmptyList[String], String] =
+        getMessages
+
+      override def inputsToList(content: String): Either[NonEmptyList[String], List[FileParameter]] =
+        getMessages
+
+      private def getMessages[T]: Either[NonEmptyList[String], T] =
+        Left(NonEmptyList.fromList(errorMessages).get)
```
Comment (myazinn) [inline services/src/test/scala/cromwell/pipeline/service/impls/WomToolTestImpl.scala]:
```suggestion
        Left(NonEmptyList.fromListUnsafe(errorMessages))
```

File: services/src/test/scala/cromwell/pipeline/service/RunServiceTest.scala [hunk 3]
```diff
@@ -41,40 +33,47 @@ class RunServiceTest extends AsyncWordSpec with Matchers with MockitoSugar {
...
+
+    "addRun" should {
       "returns run id" taggedAs Service in {
         val runCreateRequest = RunCreateRequest(projectVersion = run.projectVersion, results = run.results)
-        when(projectService.getUserProjectById(projectId, userId)).thenReturn(Future.successful(project))
-        when(runRepository.addRun(any[Run])).thenReturn(Future.successful(runId))
 
         runService.addRun(runCreateRequest, projectId, userId).map {
-          _ shouldBe runId
+          _ shouldBe a[RunId]
```
Comment (myazinn) [inline services/src/test/scala/cromwell/pipeline/service/RunServiceTest.scala]: Obviously it will be `a[RunId]`, that's what the type system will enforce even without tests. Why did you need to change this assertion?

**Review by myazinn — CHANGES_REQUESTED** [review]

File: services/src/test/scala/cromwell/pipeline/service/RunServiceTest.scala (hunk 3 above)
Comment (Nikolai-Isaev) [reply services/src/test/scala/cromwell/pipeline/service/RunServiceTest.scala]: In that case `repository.addRun` returns a random run Id, generated inside `runService.addRun`.
Can't think of the decent way of how to test it with current repository testImpl.

File: services/src/test/scala/cromwell/pipeline/service/impls/WomToolTestImpl.scala (hunk 2 above)
Comment (Nikolai-Isaev) [reply services/src/test/scala/cromwell/pipeline/service/impls/WomToolTestImpl.scala]: Fixed

File: services/src/test/scala/cromwell/pipeline/service/RunServiceTest.scala (hunk 3 above)
Comment (myazinn) [reply services/src/test/scala/cromwell/pipeline/service/RunServiceTest.scala]: In this case, there are two ways to fix is.
1) Make your assertion more explicit 
```suggestion
          _ => succeed
```
2) Change the test order and modify this one. You should move tests for `getRunByIdAndUser` method on top and then add a test for `addRun` method. This test will add a run, then fetch it with `getRunByIdAndUser` and compare that this is the same run as we expected.

Comment (Nikolai-Isaev) [reply services/src/test/scala/cromwell/pipeline/service/RunServiceTest.scala]: Great ideas, thanks! Fixed

**Review by myazinn — APPROVED** [review]

**Review by GrigoriiBerezin — APPROVED** [review]
